model/job.go

```diff
@@ -240,20 +287,45 @@ func (j *Job) MergeSpec(otherJob *Job) {
 	j.Properties = append(j.Properties, otherJob.Properties...)
 }
 
+type configConsumes struct {
```

Comment:
This is effectively the same type as `jobLinkProvider`.  Maybe there is a better name that describes this type that covers both uses? What about `jobLink`, or `linkTarget`?